module-03-react/src/components/Gallery.jsx

```diff
@@ -0,0 +1,146 @@
+                            const { results, next } = response.data;
+                            allResults = [...allResults, ...results];
+                            nextPage = next;
+                        } catch (error) {
+                            nextPage = null;
+                        }
+                    }
+                    setAllResults(allResults);
+                    setCards(allResults.slice(0, 15));
+                    if (allResults.length <= 15 && bottomButton) {
+                        bottomButton.classList.add("is-hidden");
+                    } else {
+                        bottomButton.classList.remove("is-hidden");
+                    }
+                    setSearch("");
+                } catch (error) {
+                    setError("Error: can't get data.");
+                } finally {
+                    setLoading(false);
+                }
+            };
+            getAPI();
+            setPage(1);
+        }, [url]);
+
+        useImperativeHandle(ref, () => ({
+            loadMore() {
+                cardCounter = page + 1;
+                setCards(allResults.slice(0, cardCounter * 15));
+                setPage(cardCounter);
+                if (cardCounter * 15 >= allResults.length) {
+                    bottomButton.classList.add("is-hidden");
+                } else {
+                    bottomButton.classList.remove("is-hidden");
+                }
+            },
+        }));
+        return (
+            <section className="gallery">
+                <div className="gallery__grid">
+                    {loading && <div className="loading">Loading...</div>}
+                    {!loading && !error &&
+                        allResults
+                            .filter((card) => {
+                                const searchField = url !== "films/" ? card.name : card.title;
+                                return searchField
+                                    ? search.toLowerCase() === "" ||
+                                    searchField.toLowerCase().includes(search.toLowerCase())
+                                    : false;
+                            })
+                            .slice(0, page * 15)
+                            .map((card, index) => {
+                                const isFilm = url === "films/";
+                                const cardName = isFilm ? card.title : card.name;
+                                const img = cardName ? getMainImage(cardName) : "Unknown";
+                                return (
+                                    <Card
+                                        key={index}
+                                        more={url == "people/"}
+                                        imgPromise={img}
+                                        name={cardName ? cardName.toLowerCase() : "Unknown"}
+                                        onClick={() => {
```

Comment:
Creating new arrow functions within render methods can lead to performance issues by creating new function instances on each render. This causes unnecessary re-renders of child components when React's prop comparison detects changes. Consider extracting this logic into a memoized handler